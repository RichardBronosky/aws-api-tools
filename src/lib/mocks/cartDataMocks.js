export const mockCartItem1 = {
  activeAddons: null,
  price: 279.99,
  product: {
    title: 'Adaire Brown Side Table',
    sku: '23122360',
    price: 279.99,
    list_price: 349.99,
    delivery_type: 'O',
    category: 'tables',
    free_shipping: true,
    on_sale: true,
    primary_image:
      'http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090',
    shipping_cost_code: { FL: 'VD71', SE: 'VD71', TX: 'VD71' },
    strikethrough: { FL: true, OM: true, SE: true, TX: true },
  },
  sku: '23122360',
  quantity: 1,
}

export const mockCartItem2 = {
  sku: '1141677P',
  product: {
    title: 'Laney Park Dark Gray 7 Pc Sectional',
    sku: '1141677P',
    price: 1199.99,
    list_price: 1299.99,
    delivery_type: 'D',
    category: 'livingroom',
    free_shipping: false,
    on_sale: true,
    primary_image:
      'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
    package_skus: {
      SE: [
        { sku: '12216760', quantity: 1, price: '208.43' },
        { sku: '12116768', quantity: 4, price: '172.94' },
        { sku: '12316762', quantity: 1, price: '231.63' },
        { sku: '12516766', quantity: 1, price: '68.17' },
      ],
      TX: [
        { sku: '12316762', quantity: '1', price: '231.63' },
        { sku: '12216760', quantity: '1', price: '208.43' },
        { sku: '12516766', quantity: '1', price: '68.17' },
        { sku: '12116768', quantity: '4', price: '172.94' },
      ],
      FL: [
        { sku: '12316762', quantity: '1', price: '248.65' },
        { sku: '12116768', quantity: '4', price: '185.62' },
        { sku: '12216760', quantity: '1', price: '223.71' },
        { sku: '12516766', quantity: '1', price: '73.16' },
      ],
    },
    shipping_cost_code: { FL: 'ROOM', SE: 'ROOM', TX: 'ROOM' },
  },
  price: 1199.99,
  activeAddons: null,
  quantity: 3,
}

export const mockCart = {
  cartItems: [mockCartItem1, mockCartItem2],
}
