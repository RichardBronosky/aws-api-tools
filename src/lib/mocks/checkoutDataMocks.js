export const order1 = {
  isPickup: false,
  isExpress: false,
  submitted: false,
  emailSubscribed: true,
  additionalDirections: 'test',
  lineItems: [
    {
      sku: '26043632',
      quantity: 2,
      unitPrice: 174,
      title: 'Collins Point Brown Accent Chair',
      deliveryType: 'O',
      category: 'livingroom',
      childItems: [],
    },
    {
      sku: '26043633',
      quantity: 2,
      unitPrice: 174,
      title: 'Collins Point Brown Accent Chair2',
      deliveryType: 'D',
      category: 'livingroom',
      childItems: [],
    },
    {
      sku: '26043634',
      quantity: 2,
      unitPrice: 174,
      title: 'Collins Point Brown Accent Chair3',
      deliveryType: 'U',
      category: 'livingroom',
      childItems: [],
    },
    {
      category: 'gift-card',
      childItems: [],
      deliveryType: 'T',
      quantity: 1,
      sku: '83333333',
      title: 'Gift Card',
      unitPrice: 100,
    },
  ],
  createdAt: '2019-05-29T15:20:15.170Z',
  lastModified: '2019-05-29T17:00:09.305Z',
  calendarType: 'daily',
  dDeliveryCharge: 0,
  deliveryCalendar: [],
  kDeliveryCharge: 0,
  paymentInfo: [
    {
      authorized: false,
      paymentType: 'CYBERV2',
      authorizedAmount: 377.58,
      paymentProperties: {
        postback: {
          req_card_type: '001',
          decision: 'ACCEPT',
        },
      },
    },
  ],
  pickupCalendar: [],
  pickupWarehouse: '',
  promotions: {
    cartTotal: 348,
    listPriceTotal: 348,
    salePriceTotal: 348,
    bonusBuyTotal: 0,
    totalSavings: 0,
    lineItems: [
      {
        sku: '26043632',
        quantity: 2,
        unitPrice: 174,
        title: 'Collins Point Brown Accent Chair',
        deliveryType: 'O',
        category: 'livingroom',
        childItems: [],
      },
    ],
    promotions: [],
  },
  total: 377.58,
  totalDeliveryCharge: 0,
  contact: {
    firstName: 'Tester',
    lastName: 'Test',
    email: 'test@test.com',
    altPhone: '',
    phone: '4040404040',
  },
  payer: {
    firstName: 'Tester',
    lastName: 'Test',
    email: 'test@test.com',
    altPhone: '',
    phone: '4040404040',
    billingDifferent: true,
    billingSubmitted: false,
  },
  deliveryTexts: true,
  distributionIndex: 1,
  emailCampaign: true,
  shippingAddress: {
    address1: '103 seffner ave',
    address2: '',
    city: 'seffner',
    state: 'fl',
    zip: '33584',
    county: 'Hillsborough',
    country_code: 'US',
    addressLookup: '',
    addressLookupSuccess: false,
    showAddressLookup: false,
  },
  billingAddress: {
    address1: '103 seffner ave',
    address2: '',
    city: 'seffner',
    state: 'fl',
    zip: '33584',
    county: 'Hillsborough',
    country_code: 'US',
    addressLookup: '',
    addressLookupSuccess: false,
    showAddressLookup: false,
  },
  additionalDirections: '',
  deliveryDate: '2019-06-04',
  shouldCombineUPSWithTruck: true,
  tax: 29.58,
  amountDue: 0,
  cyberSourceSignature: {
    bill_to_forename: 'Tester',
    bill_to_surname: 'Test',
    bill_to_email: 'test@test.com',
    bill_to_address_line1: '103 seffner ave',
    bill_to_address_city: 'seffner',
    bill_to_address_postal_code: '33584',
    bill_to_address_state: 'FL',
    bill_to_address_country: 'US',
    amount: '377.58',
    currency: 'USD',
    reference_number: '9d06c1c1-ea10-4b07-82e0-27ae4947e6ef',
    locale: 'en-us',
    signed_date_time: '2019-05-29T16:59:55Z',
    profile_id: 'C43A2464-4A37-4A37-BF2C-6E62CFCEBA41',
    access_key: '66674d4c4897393fb6941963f6ac6d1a',
    transaction_uuid: 'd9aa39ff-793c-4157-b017-853ed824ad82',
    transaction_type: 'create_payment_token',
    unsigned_field_names: 'payment_method',
    payment_method: 'card',
    item_0_name: '26043632',
    item_0_sku: '26043632',
    item_0_quantity: 2,
    item_0_unit_price: '377.58',
    item_0_taxAmount: '0.00',
    line_item_count: 1,
    signed_field_names:
      'bill_to_forename,bill_to_surname,bill_to_email,bill_to_address_line1,bill_to_address_city,bill_to_address_postal_code,bill_to_address_state,bill_to_address_country,amount,currency,reference_number,locale,signed_date_time,profile_id,access_key,transaction_uuid,transaction_type,unsigned_field_names,item_0_name,item_0_sku,item_0_quantity,item_0_unit_price,item_0_taxAmount,line_item_count,signed_field_names',
    signature: 'T0y65CK+W73H2uk483GgOHGVWSQmsPSZs3Hi1RR0NKo=',
  },
  giftCardInfo: {
    giftCards: [],
    useGiftCard: false,
    giftCardNumber: '',
    giftCardPin: '',
    addAnotherCard: false,
  },
  orderId: '9d06c1c1-ea10-4b07-82e0-27ae4947e6ef',
}

export const mockDeliveryCalendar = [
  {
    date: '2019-06-04',
    readable: 'June 04',
    dayOfWeek: 'Tue',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
  {
    date: '2019-06-05',
    readable: 'June 05',
    dayOfWeek: 'Wed',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
  {
    date: '2019-06-06',
    readable: 'June 06',
    dayOfWeek: 'Thu',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
  {
    date: '2019-06-07',
    readable: 'June 07',
    dayOfWeek: 'Fri',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
]

export const fetchAddressLookupResp = {
  totalMatches: 136009,
  count: 9,
  results: [
    {
      suggestion: '4100 Alexandria Pike, Alexandria KY 41001',
      matched: [[36, 40], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=15197f6d-d4b7-41ad-bc77-b0d5790baa0a_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Amy Lou Dr, Alexandria KY 41001',
      matched: [[31, 35], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=420547b1-8fac-4d4d-bc9c-776fc6a7b6ff_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Beiting Dr, Alexandria KY 41001',
      matched: [[31, 35], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=61f7eaae-0ad7-4f7f-8cbe-68e126ad190e_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Ky 915, Alexandria KY 41001',
      matched: [[27, 31], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=204b8965-23a8-4d68-a296-5c17e03cd105_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Licking Pike, Alexandria KY 41001',
      matched: [[33, 37], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=ed19c1cb-ce4d-47d8-b967-50f3f43cd3b0_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Old State Road 2, Alexandria KY 41001',
      matched: [[37, 41], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=f67f813f-97e5-4683-9274-579267c53238_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Persimmon Grove Rd, Alexandria KY 41001',
      matched: [[39, 43], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=5cb5e9d7-8cd5-49e0-b2eb-be92c95ca940_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Sr 10, Alexandria KY 41001',
      matched: [[26, 30], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=4edb7aa7-527d-4c0d-a4c7-5534b858e7ac_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 State Route 10, Alexandria KY 41001',
      matched: [[35, 39], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=90b44538-006a-488a-bf80-6e0517808017_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
  ],
}

export const giftCards1 = [
  {
    giftCardNumber: '1111111111111111',
    giftCardPin: '111111',
    authorizedAmount: 100,
    balance: 200,
  },
]
