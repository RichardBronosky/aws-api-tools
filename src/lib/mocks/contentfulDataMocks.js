export const Banner1 = {
  contentfulContentGroup: {
    groupContent: [
      {
        __typename: 'ContentfulBanner',
        title: 'Promo E : Umbrella Sale 1 - C6 R1',
        startDate: null,
        endDate: null,
        displayContent: true,
        contentText: {
          childMarkdownRemark: {
            html:
              '<br>\n<span style="color:#FFFFFF; font-weight: bold; font-size: 28px;">GET A FREE UMBRELLA</span>\n<br>\n<span style="color:#FFFFFF; font-weight: bold; font-size: 28px;">WITH SELECT* OUTDOOR PURCHASES OF $999+</span>\n<br>\n<br>\n<span style="color:#FFFFFF; font-weight: bold;">*EXCLUDES VENDOR DIRECT.</span>\n<br>',
          },
        },
        contentHorizontalRuleColor: null,
        textPosition: 'center',
        contentPosition: 'center',
        containerSize: '50%',
        contentLinkColor: 'white',
        link: {
          id: '12703942-b7d5-5bc8-b79b-ad18d5321106',
          text: 'Link : Outdoor Patio',
          displayText: 'Outdoor',
          textColor: null,
          internalUrl: null,
          slug: '/furniture/outdoor-patio',
          url: null,
          altDesc: 'View All Outdoor Patio Furniture',
          title: null,
          type: null,
          target: null,
        },
        contentLink: null,
        buttons: null,
        imageAltText: 'Shop Umbrella Sale',
        desktopBackgroundImage: null,
        backgroundColor: 'Blue',
        mobileImage: {
          id: '4e8177a3-6c27-5544-a5ee-fcb384528020',
          sizes: {
            srcSet:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png?w=640&h=600&q=50 640w',
            src:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png?q=50',
            sizes: '(max-width: undefinedpx) 100vw, undefinedpx',
          },
          file: {
            url:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png',
            fileName: 'AnnSale_Umbrella1_MOBILE_640x600.png',
            contentType: 'image/png',
          },
          resolutions: { width: null, height: null },
        },
        desktopImage: {
          id: '8740a3af-1f18-5f98-9335-137b1f18a7c7',
          sizes: {
            srcSet:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png?w=1660&h=350&q=50 1660w',
            src:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png?q=50',
            sizes: '(max-width: undefinedpx) 100vw, undefinedpx',
          },
          file: {
            url:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png',
            fileName: 'AnnSale_Umbrella1_CONTAINER_1660x350.png',
            contentType: 'image/png',
          },
          resolutions: { width: null, height: null },
        },
        contentPadding: null,
        contentful_id: '3fezOgXNiGTW87ngJIx9cB',
      },
    ],
  },
}
