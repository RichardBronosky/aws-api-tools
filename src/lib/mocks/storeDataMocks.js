export const mockStore1 = {
  id: '37ae4548-c47a-5e5d-8c42-c57143939614',
  storeNumber: 1703,
  storeName: null,
  storeType: 'Showroom',
  address1: '10845 HAYNES BRIDGE RD',
  address2: null,
  city: 'Alpharetta',
  state: 'GA',
  zip: '30022',
  location: { lat: 34.0480313, lon: -84.28556530000003 },
  phoneNumber: '770-664-0100',
  storeHours: {
    mondayOpen: '10:00',
    mondayClosed: '21:00',
    tuesdayOpen: '10:00',
    tuesdayClosed: '21:00',
    wednesdayOpen: '10:00',
    wednesdayClosed: '21:00',
    thursdayOpen: '10:00',
    thursdayClosed: '21:00',
    fridayOpen: '10:00',
    fridayClosed: '21:00',
    saturdayOpen: '10:00',
    saturdayClosed: '21:00',
    sundayOpen: '10:00',
    sundayClosed: '21:00',
  },
  pickup_hours: {
    mondayOpen: '',
    mondayClosed: '',
    tuesdayOpen: '',
    tuesdayClosed: '',
    wednesdayOpen: '10:00',
    wednesdayClosed: '21:00',
    thursdayOpen: '10:00',
    thursdayClosed: '21:00',
    fridayOpen: '10:00',
    fridayClosed: '21:00',
    saturdayOpen: '10:00',
    saturdayClosed: '21:00',
    sundayOpen: '',
    sundayClosed: '',
  },
  neighborhoodsNearby: ['Berkely Lake', 'Cumming', 'Duluth', 'Dunwoody', 'Mountain Park', 'Nocross', 'Roswell'],
  specialClosings: null,
}

export const mockWarehouse1 = {
  storeNumber: 1,
  id: '99',
  division: 'FL',
  hoursType: 'P',
  address1: '1475 AIRPORT RD',
  address2: '',
  city: 'Lakeland',
  closeDate: '',
  pickup_hours: {
    mondayOpen: '',
    mondayClosed: '',
    tuesdayOpen: '',
    tuesdayClosed: '',
    wednesdayOpen: '10:00',
    wednesdayClosed: '21:00',
    thursdayOpen: '10:00',
    thursdayClosed: '21:00',
    fridayOpen: '10:00',
    fridayClosed: '21:00',
    saturdayOpen: '10:00',
    saturdayClosed: '21:00',
    sundayOpen: '',
    sundayClosed: '',
  },
  latitude: 28.023763,
  longitude: -82.017496,
  openDate: '6/29/99 0:00',
  phone: '863/686-3737',
  state: 'FL',
  statusTypeId: 'A',
  storeTypeId: 'W',
  zip: 33811,
}

export const mockDistirbutionCenter = {
  name: `Lakeland Distribution Center`,
  hours: [
    {
      closeTime: '',
      day: 'Sunday',
      openTime: '',
    },
    {
      closeTime: '',
      day: 'Monday',
      openTime: '',
    },
    {
      closeTime: '',
      day: 'Tuesday',
      openTime: '',
    },
    {
      closeTime: '21:00',
      day: 'Wednesday',
      openTime: '10:00',
    },
    {
      closeTime: '21:00',
      day: 'Thursday',
      openTime: '10:00',
    },
    {
      closeTime: '21:00',
      day: 'Friday',
      openTime: '10:00',
    },
    {
      closeTime: '21:00',
      day: 'Saturday',
      openTime: '10:00',
    },
  ],
}
