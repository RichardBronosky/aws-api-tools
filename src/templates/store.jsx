import React from 'react'
import Helmet from 'react-helmet'
import { Map, Marker, InfoWindow, GoogleApiWrapper } from 'google-maps-react'
import Layout from '../components/layout'
import Breadcrumb from '../components/shared/breadcrumbs'
import RTGLink from '../components/shared/link'
import { decodeHtml, slugify, weekdays, months, getStandardTime } from '../lib/helpers/string-helper'
import { contentfulImage } from '../lib/helpers/contentful'
import { getStoreHours, getStoreName, getStoreInfoFromFeed } from '../lib/helpers/store-locator'
import { expandState } from '../lib/helpers/geo-location'
import { setupAnalytics } from '../lib/helpers/google-tag-manager'
import StoreInfo from '../components/store-locator/store-info'
import '../assets/css/pages/store-location.sass'

class Store extends React.Component {
  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedMarker: {},
  }

  componentDidMount() {
    const store = this.props.pageContext.store
    setupAnalytics({ pageData: { type: 'store', title: getStoreName(store), path: window.location.pathname } })
  }

  createCrumb = (crumb, slugPath) => {
    return {
      altDesc: `View All ${ crumb } Products`,
      id: slugPath,
      slug: slugPath,
      text: crumb,
    }
  }

  generateStoreCrumbs = store => {
    let crumbs = []
    crumbs.push(this.createCrumb('Find a Showroom', '/stores'))
    if (store.state) {
      crumbs.push(
        this.createCrumb(
          `Find a Showroom - ${ expandState(store.state) }`,
          `/stores/${ slugify(expandState(store.state)) }`
        )
      )
    }
    crumbs.push(
      this.createCrumb(
        decodeHtml(`${ store.city }, ${ store.state }${ store.storeName ? ` (${ store.storeName })` : '' }`),
        window.location.pathname
      )
    )
    return {
      crumbs: crumbs,
    }
  }

  onLocationClick = (props, marker) => {
    const store = this.props.pageContext.store
    const storeInfoFromFeed = getStoreInfoFromFeed(store.storeNumber)
    let fullMarker = {
      ...props,
      title: store.address1,
      name: getStoreName(store),
      lat: store.location.lat,
      lng: store.location.lon,
      city: store.city,
      state: store.state,
      storeNumber: store.storeNumber,
      zip: store.zip,
      phone: storeInfoFromFeed.phone.replace('/', '-'),
      hours: getStoreHours(storeInfoFromFeed.store_hours),
    }
    this.setState({
      selectedMarker: fullMarker,
      activeMarker: marker,
      showingInfoWindow: true,
    })
  }

  onMapClicked = () => {
    if (this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
      })
    }
  }

  getTodayOpen = store => {
    const currentDay = weekdays[new Date().getDay()]
    const month = months[new Date().getMonth()].substr(0, 3)
    const day = new Date().getDate()
    const storeOpen = store.store_hours[`${ currentDay.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ currentDay.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ currentDay }, ${ month } ${ day }: ${
        storeOpen ? `Open from ${ getStandardTime(storeOpen) } to ${ getStandardTime(storeClose) }` : 'Closed'
      }`
    } else {
      ;`${ currentDay }, ${ month } ${ day }`
    }
  }

  getHours = (store, day) => {
    const storeOpen = store.store_hours[`${ day.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ day.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ getStandardTime(storeOpen) } - ${ getStandardTime(storeClose) }`
    }
  }

  render() {
    const { activeMarker, selectedMarker, showingInfoWindow } = this.state
    const store = this.props.pageContext.store
    const address = `${ store.address1 }${ store.address2 ? `, ${ store.address2 }` : '' }, ${ store.city }, ${ store.state } ${
      store.zip
    } `
    const storeInfoFromFeed = getStoreInfoFromFeed(store && store.storeNumber)
    if (store) {
      let storeTypeTitle = `${ store.city }, ${ store.state }`
      if (store.storeType === 'Showroom' || store.storeType === 'Express') {
        storeTypeTitle = storeTypeTitle + ' Furniture & Mattress Store'
      } else if (store.storeType === 'Kids') {
        storeTypeTitle = storeTypeTitle + ' Kids Furniture & Mattress Store'
      } else if (store.storeType === 'Outlet') {
        storeTypeTitle = storeTypeTitle + ' Affordable Furniture Outlet Store'
      } else if (store.storeType === 'Outdoor') {
        storeTypeTitle = storeTypeTitle + ' Patio Furniture: Wicker, Teak Outdoor Furniture Sets'
      }
      return (
        <Layout { ...this.props }>
          <Helmet title={ storeTypeTitle } />
          { store.seo && (
            <>
              { store.seo.pageTitle && <Helmet title={ store.seo.pageTitle } /> }
              { store.seo.canonical && (
                <Helmet
                  link={ [
                    {
                      rel: 'canonical',
                      href: store.seo.canonical,
                    },
                  ] }
                />
              ) }
              { store.seo.metaDescription && (
                <Helmet
                  meta={ [
                    {
                      name: 'description',
                      content: store.seo.metaDescription.metaDescription,
                    },
                  ] }
                />
              ) }
              { store.seo.metaRobots && (
                <Helmet
                  meta={ [
                    {
                      name: 'robots',
                      content: store.seo.metaRobots,
                    },
                  ] }
                />
              ) }
              { store.seo.jsonLdSchema && (
                <Helmet
                  script={ [
                    {
                      type: 'application/ld+json',
                      innerHTML: store.seo.jsonLdSchema.jsonLdSchema,
                    },
                  ] }
                />
              ) }
            </>
          ) }
          <section className="seo-page-heading full-width hide show">
            <div className="grid-container ">
              <div className="grid-x grid-margin-x">
                <div className="cell small-12">
                  <h1>
                    { store.seo && store.seo.pageHeading
                      ? store.seo.pageHeading
                      : `${ storeTypeTitle }${ store.storeName ? ` (${ store.storeName })` : '' }` }
                  </h1>
                </div>
              </div>
            </div>
          </section>
          <div className="stores-breadcrumb">
            <Breadcrumb data={ this.generateStoreCrumbs(store) } includeHeading={ true } />
          </div>
          <div className="store-location">
            <div className="loc-info">
              { address }
              <RTGLink
                data={ {
                  url: `tel:${ storeInfoFromFeed.phone.replace('/', '-') }`,
                  title: 'Call store',
                  category: 'store page',
                  action: 'Call click',
                  label: 'Call',
                } }
                className="phone-link"
              >
                { storeInfoFromFeed.phone.replace('/', '-') }
              </RTGLink>
              <span className="today">{ this.getTodayOpen(storeInfoFromFeed) }</span>
            </div>
            <div className="grid-x map-container">
              <Map
                initialCenter={ { lat: store.location.lat, lng: store.location.lon } }
                center={ { lat: store.location.lat, lng: store.location.lon } }
                className="map"
                google={ this.props.google }
                zoom={ 11 }
                onClick={ this.onMapClicked }
              >
                <Marker
                  position={ { lat: store.location.lat, lng: store.location.lon } }
                  title={ `${ store.city }${ store.storeName ? ` - ${ store.storeName }` : '' } ${ store.storeType }` }
                  icon={ {
                    url: `http://maps.google.com/mapfiles/markerA.png`,
                  } }
                  onClick={ this.onLocationClick }
                />
                <InfoWindow marker={ activeMarker } visible={ showingInfoWindow }>
                  <StoreInfo selectedMarker={ selectedMarker } />
                </InfoWindow>
              </Map>
              { store.storeImage && (
                <div className="store-image-container">
                  <img className="store-image" alt={ storeTypeTitle } src={ contentfulImage(store.storeImage.file.url) } />
                </div>
              ) }
              { storeInfoFromFeed.store_hours && (
                <div className="section-container">
                  <h2 className="section-heading">Store Hours:</h2>
                  { weekdays.map(day => (
                    <div className="hour" key={ day }>
                      { storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">{ this.getHours(storeInfoFromFeed, day) }</span>
                        </>
                      ) }
                      { !storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">Closed</span>
                        </>
                      ) }
                    </div>
                  )) }
                  { store.specialClosings && (
                    <span
                      className="special-closings"
                      dangerouslySetInnerHTML={ { __html: store.specialClosings.markdown.childMarkdownRemark.html } }
                    />
                  ) }
                </div>
              ) }
              { store.directions && (
                <div className="section-container">
                  <h2 className="section-heading">Store Location / Address:</h2>
                  <span>{ address }</span>
                  <br />
                  <span
                    className="directions"
                    dangerouslySetInnerHTML={ { __html: store.directions.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.neighborhoodsNearby && store.neighborhoodsNearby.length > 0 && (
                <div className="section-container">
                  <h2 className="section-heading">Neighborhoods Nearby:</h2>
                  <ul>
                    { store.neighborhoodsNearby.map((hood, index) => (
                      <li key={ index } className="hood">
                        { hood }
                      </li>
                    )) }
                  </ul>
                </div>
              ) }
              { store.description && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">Store Description:</h2>
                  <span
                    className="description"
                    dangerouslySetInnerHTML={ { __html: store.description.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.faqs && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">FAQs:</h2>
                  <span
                    className="faqs"
                    dangerouslySetInnerHTML={ { __html: store.faqs.markdown.childMarkdownRemark.html } }
                  />
                </div>
              ) }
            </div>
          </div>
        </Layout>
      )
    } else {
      return null
    }
  }
}

export default GoogleApiWrapper({
  apiKey: `${ process.env.GATSBY_GOOGLE_BROWSER_API_KEY }`,
})(Store)
