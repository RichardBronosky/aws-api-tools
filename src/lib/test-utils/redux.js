import configureMockStore from 'redux-mock-store'
import * as reduxStore from '../../redux/store'

const mockStore = configureMockStore([])

export const mockStoreDispatch = initialState => {
  reduxStore.store = mockStore(initialState)
  reduxStore.store.dispatch = jest.fn()
}
