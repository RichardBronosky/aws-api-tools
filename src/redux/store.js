import React from 'react'
import { Provider } from 'react-redux'
import { createStore, compose, combineReducers } from 'redux'

import global from './modules/global'
import location from './modules/location'
import cart from './modules/cart'
import checkout from './modules/checkout'

const rootReducer = combineReducers({
  global,
  location,
  cart,
  checkout,
})

// Redux DevTools
const windowObject = typeof window !== 'undefined' && window
const reduxDevTools =
  process.env.NODE_ENV === 'development' && windowObject.devToolsExtension
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : _ => _

const configureStore = initialState => createStore(rootReducer, initialState, compose(reduxDevTools))

export const store = configureStore()

// eslint-disable-next-line react/display-name,react/prop-types
export default ({ element }) => <Provider store={ store }>{ element }</Provider>
