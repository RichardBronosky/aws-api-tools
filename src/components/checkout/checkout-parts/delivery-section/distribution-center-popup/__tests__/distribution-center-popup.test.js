import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '../../../../../../redux/store'
import DistributionCenterPopup from '../distribution-center-popup'
import { mockWarehouse1 } from '../../../../../../lib/mocks/storeDataMocks'

describe('DistributionCenterPopup', () => {
  it('renders correctly when popup is open', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DistributionCenterPopup store={ mockWarehouse1 } open />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when popup is closed', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <DistributionCenterPopup store={ mockWarehouse1 } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
