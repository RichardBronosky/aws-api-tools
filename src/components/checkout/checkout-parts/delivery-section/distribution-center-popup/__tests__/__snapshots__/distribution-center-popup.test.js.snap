// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistributionCenterPopup renders correctly when popup is closed 1`] = `
<div
  className="distribution-center-popup"
>
  <button
    alt="Lakeland Distribution Center"
    className="distribution-center-button"
    onClick={[Function]}
    tabIndex="0"
  >
    Lakeland Distribution Center
  </button>
</div>
`;

exports[`DistributionCenterPopup renders correctly when popup is open 1`] = `
<div
  className="distribution-center-popup"
>
  <button
    alt="Lakeland Distribution Center"
    aria-expanded={true}
    className="distribution-center-button"
    onClick={[Function]}
    tabIndex="0"
  >
    Lakeland Distribution Center
  </button>
  <div
    className="popup container"
  >
    <p>
      1475 AIRPORT RD, Lakeland, FL 33811
    </p>
    <div>
      Contact:
      <a
        gtm-action="Call distribution center click"
        gtm-category="cart/checkout"
        gtm-label="Call distribution center"
        gtm-value={null}
        href="tel:863-686-3737"
        onClick={[Function]}
        title="Call distribution center"
      >
        863-686-3737
      </a>
    </div>
  </div>
</div>
`;
